src/main.ts

```diff
@@ -322,15 +322,20 @@ export default class LinterPlugin extends Plugin {
     this.originalSaveCallback = saveCommandDefinition?.checkCallback;
 
     if (typeof this.originalSaveCallback === 'function') {
-      saveCommandDefinition.checkCallback = () => {
-        this.originalSaveCallback();
+      saveCommandDefinition.checkCallback = (checking: boolean) => {
+
+        if (checking) {
+          return this.originalSaveCallback(checking);
+        } else {
+          this.originalSaveCallback(checking)
 
-        if (this.settings.lintOnSave && this.isEnabled) {
           const editor = this.getEditor();
-          if (editor) {
-            const file = this.app.workspace.getActiveFile();
-            if (!this.shouldIgnoreFile(file) && this.isMarkdownFile(file) && editor.cm) {
-              void this.runLinterEditor(editor);
+          const file = this.app.workspace.getActiveFile();
+          if (this.settings.lintOnSave && this.isEnabled) {
+            if (editor) {
+              if (!this.shouldIgnoreFile(file) && this.isMarkdownFile(file) && editor.cm) {
+                void this.runLinterEditor(editor);
+              }
```

Comment:
The logic here should be the same as the original to avoid trying to act on null or undefined files:
``` typescript
 if (this.settings.lintOnSave && this.isEnabled) {
          const editor = this.getEditor();
          if (editor) {
            const file = this.app.workspace.getActiveFile();
            if (!this.shouldIgnoreFile(file) && this.isMarkdownFile(file) && editor.cm) {
              if (!checking) {
                this.runLinterEditor(editor);
              }
              return true;
            }
          }
        }
      };
```